packages/nimble-components/src/table/specs/table-column-specs/table-column-file-size.md

```diff
@@ -0,0 +1,222 @@
+# File Size Table Column
+
+## Overview
+
+The file size table column will provide a way to visualize numeric data that represents a file size (or other binary data size). It will automatically convert the raw value to use an appropriate display unit.
+
+### Background
+
+[Work Item](https://ni.visualstudio.com/DevCentral/_workitems/edit/2526114)
+
+### Non-goals
+
+-   Configurable number of decimal places
+-   Configurable units (e.g. display all values in KB)
+-   Grouping by value ranges/buckets
+
+### Features
+
+-   Converts the byte value to largest unit (e.g. KB, MB, GB) that results in a value of >=1
+-   Option to use 1024-based units (e.g. KiB) or 1000-based units (e.g. KB)
+-   Displays up to one decimal place
+-   Sorts/groups by numeric value (number of bytes) rather than display string
+-   Localized number and unit strings
+
+### Risks and Challenges
+
+### Prior Art/Examples
+
+Size column in [SLE Files grid](https://dev.lifecyclesolutions.ni.com/files/)
+
+![SLE Files grid](../spec-images/fileSizesSle.png)
+
+Windows explorer (details view)
+
+![Windows explorer](../spec-images/fileSizesWindows.png)
+
+Chrome Dev Tools Network tab
+
+![Chrome Dev Tools Network tab](../spec-images/fileSizeChromeDevTools.png)
+
+---
+
+## Design
+
+### API
+
+`nimble-table-column-file-size`
+
+-   `field-name` - name of the record field containing the data. Data must be a `number` byte count.
+-   `unit-type` - "binary" (KiB, MiB, GiB, TiB, PiB) or "decimal" (KB, MB, GB, TB, PB); defaults to "decimal"
+
+The component will extend `TableColumnTextBase`, thereby including the APIs for a **groupable** and **fractional-width** column.
+
+### Examples
+
+Given a data value of 2856 (bytes):
```

Comment:
<nit> The grokking of what is going on here might be helped if the value was simply 1024 (or some integer multiple of that).